File: leetcode/easy/arranging-coins/arranging-coins.js
/**
 * @param {number} n
 * @return {number}
 */
var arrangeCoins = function(n) {
  let rowCount = 0;
  while (n > 0) {
      rowCount++;
      n -= rowCount;
  }
  return n == 0 ? rowCount : rowCount - 1;
};

File: leetcode/easy/arranging-coins/arranging-coins.java
class Solution {
    public int arrangeCoins(int n) {
        if (n == 1) return n;

        int rowCount = 1;
        while (n > 0) {
            n -= rowCount;
            if (n > 0) rowCount++;
        }

        return n == 0 ? rowCount : rowCount - 1;
    }
}